lib/Doctrine/ODM/PHPCR/ReferenceManyCollection.php

```diff
@@ -96,7 +99,13 @@ public function initialize()
     {
         if (!$this->isInitialized()) {
             $referencedDocs = array();
-            $referencedNodes = $this->dm->getPhpcrSession()->getNodesByIdentifier($this->referencedNodes);
+            if ($this->referenceType == 'path') {
```

Comment:
please have the constant value on the left side and use identity comparison:
`if (static::REFERENCE_TYPE_PATH === $this->referenceType) {`